Tests/UtilityTests/PkgConfigParserTests.swift

```diff
@@ -69,17 +69,27 @@ final class PkgConfigParserTests: XCTestCase {
             "/usr/lib/pkgconfig/foo.pc",
             "/usr/local/opt/foo/lib/pkgconfig/foo.pc",
             "/custom/foo.pc")
-        XCTAssertEqual("/custom/foo.pc", try PCFileFinder(diagnostics: diagnostics).locatePCFile(name: "foo", customSearchPaths: [AbsolutePath("/custom")], fileSystem: fs).asString)
-        XCTAssertEqual("/custom/foo.pc", try PkgConfig(name: "foo", additionalSearchPaths: [AbsolutePath("/custom")], diagnostics: diagnostics, fileSystem: fs).pcFile.asString)
-        XCTAssertEqual("/usr/lib/pkgconfig/foo.pc", try PCFileFinder(diagnostics: diagnostics).locatePCFile(name: "foo", customSearchPaths: [], fileSystem: fs).asString)
+        XCTAssertEqual("/custom/foo.pc", try PCFileFinder(diagnostics: diagnostics, fileSystem: fs, brewPrefix: nil).locatePCFile(name: "foo", customSearchPaths: [AbsolutePath("/custom")]).asString)
+        XCTAssertEqual("/custom/foo.pc", try PkgConfig(name: "foo", additionalSearchPaths: [AbsolutePath("/custom")], diagnostics: diagnostics, fileSystem: fs, brewPrefix: nil).pcFile.asString)
+        XCTAssertEqual("/usr/lib/pkgconfig/foo.pc", try PCFileFinder(diagnostics: diagnostics, fileSystem: fs, brewPrefix: nil).locatePCFile(name: "foo", customSearchPaths: []).asString)
         try withCustomEnv(["PKG_CONFIG_PATH": "/usr/local/opt/foo/lib/pkgconfig"]) {
-            XCTAssertEqual("/usr/local/opt/foo/lib/pkgconfig/foo.pc", try PkgConfig(name: "foo", diagnostics: diagnostics, fileSystem: fs).pcFile.asString)
+            XCTAssertEqual("/usr/local/opt/foo/lib/pkgconfig/foo.pc", try PkgConfig(name: "foo", diagnostics: diagnostics, fileSystem: fs, brewPrefix: nil).pcFile.asString)
         }
         try withCustomEnv(["PKG_CONFIG_PATH": "/usr/local/opt/foo/lib/pkgconfig:/usr/lib/pkgconfig"]) {
-            XCTAssertEqual("/usr/local/opt/foo/lib/pkgconfig/foo.pc", try PkgConfig(name: "foo", diagnostics: diagnostics, fileSystem: fs).pcFile.asString)
+            XCTAssertEqual("/usr/local/opt/foo/lib/pkgconfig/foo.pc", try PkgConfig(name: "foo", diagnostics: diagnostics, fileSystem: fs, brewPrefix: nil).pcFile.asString)
         }
     }
 
+    func testBrewPrefix() throws {
+        let fakePkgConfig = AbsolutePath("/Volumes/BestDrive/bin/pkg-config")
+        let fs = InMemoryFileSystem(emptyFiles: [fakePkgConfig.asString])
+        try fs.chmod(FileMode.executable, path: fakePkgConfig, options: [])
+
+        let diagnostics = DiagnosticsEngine()
+        _ = PCFileFinder(diagnostics: diagnostics, fileSystem: fs, brewPrefix: fakePkgConfig.parentDirectory.parentDirectory)
+        XCTAssertEqual(Process.validatedExecutablesMap, ["pkg-config": fakePkgConfig])
```

Comment:
I don't think we should do this. It feels like wrong type of assertion. We're asserting on a side-effect that we know about instead of checking input and output of the main functionality that we're testing. It is also exposing a thread unsafe variable in the process. I think we should do the pkg-config executable instead.

Reply:
Not sure I agree. We *do* want to test that a specific version of the `pkg-config` executable is found, the main functionality should be tested in other tests.

Reply:
For exposing the tread unsafe variable, we could instead restructure `Process` so that we can verify which executable is being found in a different way.

Reply:
A fake executable can return something unique that the result obtained from PCFileFinder call can be asserted on. That sounds much better to me.

Reply:
I stand corrected, launching doesn't actually work in practice (it'll still rely on `PATH`), so changing the test makes sense.